### PR TITLE
Chat: move chat to existing group

### PR DESCRIPTION
**Pull request by Fang-**
Expands on the "groupify" functionality added in #2480 by allowing you to merge invite-only chats into an existing group.

UI/copy here might need some light touching up. Currently just display the group search, lets you continue without selecting a group (old behavior, creates a new group based on the chat), or shows you the "inclusive" toggle if you do select a target group (new behavior, merges chat into existing group).

convert

### CODE REVIEW

**Comment by matildepark** [comment]
If you're going to rewrite invite search, you should update all the copies of it across all apps.

File: pkg/interface/chat/src/js/components/settings.js [hunk 1]
```diff
@@ -171,15 +191,56 @@ export class SettingsScreen extends Component {
...
+        inclusiveToggle = (
+          <div className="mt4">
+            <input
+              type="checkbox"
+              style={{ WebkitAppearance: "none", width: 28 }}
+              className={inclusiveClasses}
+              onChange={this.changeInclusive}
+            />
+            <span className="dib f9 white-d inter ml3">
+              Inclusive
```
Comment (matildepark) [inline pkg/interface/chat/src/js/components/settings.js]: `Add all members to group` and keep the body text and this is fine
```suggestion
              Add all members to group
```

File: pkg/interface/chat/src/js/components/settings.js [hunk 2]
```diff
@@ -171,15 +191,56 @@ export class SettingsScreen extends Component {
     if (!ownedUnmanagedVillage) {
       return null;
     } else {
+      let inclusiveToggle = <div/>
+      console.log('tg', state.targetGroup);
+      if (state.targetGroup) {
+        //TODO toggle component
```
Comment (matildepark) [inline pkg/interface/chat/src/js/components/settings.js]: seems like it's here

**Comment by ixv** [comment]
I tried the case where members of the chat are not in the group, and the inclusive toggle is on. The ship gets invited to the group successfully, but the old chat persists with the "your ship has been disconnected" notice. Joining the new chat manually works fine, but then trying to leave the old chat manually gives me this: 
image

**Comment by Fang-** [comment]
> the old chat persists with the "your ship has been disconnected" notice

I don't think we can (or should) push deletion in the current model. Maybe a `%moved` diff for cases like these is appropriate, but seems way out of scope.

Should we maybe send invites for the chat, in addition to invites for the group?

File: pkg/interface/chat/src/js/components/settings.js (hunk 2 above)
Comment (Fang-) [reply pkg/interface/chat/src/js/components/settings.js]: I meant like a `component/lib/toggle.js`. Will clarify.

**Comment by ixv** [comment]
Yeah, I don't have a huge problem with this. If you fix the issue with leaving those old chats, I'm fine with this going in.

**Comment by Fang-** [comment]
> If you're going to rewrite invite search, you should update all the copies of it across all apps.

Did we decide not to have a `/pkg/interface/lib` at some point, or have we just not gotten around to it yet?

@ixv this _seems_ to be a chat/metadata bug, where chat deletion pushes deletion of everything (metadata, group) but the chat itself. Those things are relied upon during local deletion, and so it crashes. I'll investigate a bit further and try to find an appropriate fix here...

**Comment by matildepark** [comment]
I think the goal is to use an npm package for that and it’s what’s in progress among the rest of interface right now. Also I’m unsure we can really reach out of the directory of the npm package like that (since we have package.json in all the apps individually). Maybe if we refactor it all into one big one? And be better with gulp? But don’t want to step on @jtobin’s jtoes...

**Comment by Fang-** [comment]
Correction: we never have the group, but we do have metadata, with a pointer to the group. I think this means the solution is to just, if the metadata isn't there, forego deleting the (non-existent) associations etc.

@ixv Please look this over carefully. Maybe the Real Solution is fixing some other problem, but I don't think this change does any harm, and at least supports deletion for the current situation.

**Comment by ixv** [comment]
Looks good, and I tested it and it works fine.

**Comment by jtobin** [comment]
> I think the goal is to use an npm package for that and it’s what’s in progress among the rest of interface right now. Also I’m unsure we can really reach out of the directory of the npm package like that (since we have package.json in all the apps individually). Maybe if we refactor it all into one big one? And be better with gulp? But don’t want to step on @jtobin’s jtoes...

I think it does make sense to refactor them into one big 'urbit-landscape' package or whatever.  I might be able to just go ahead and do this since I need to update #2349 for the post-OS1 stuff anyway.

**Pull request merged**